### PR TITLE
[FLINK-20913][hive]Improve HiveConf creation

### PR DESCRIPTION
**Pull request by dixingxing0**
## What is the purpose of the change

This pull request improves HiveConf creation. see https://issues.apache.org/jira/browse/FLINK-20913

## Brief change log
Introduce HiveConfUtils to create HiveConf instance, avoid reset hive configuration to default value.

## Verifying this change

This change is already covered by existing tests, such as *(HiveTableSourceITCase)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature?  no

### CODE REVIEW

**Comment by dixingxing0** [comment]
@flinkbot

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/util/HiveConfUtils.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package org.apache.flink.connectors.hive.util;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+/** Create HiveConf. */
+public class HiveConfUtils {
+
+    /**
+     * Create HiveConf instance via hadoop configuration.
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/util/HiveConfUtils.java:28]: Please add more comments to explain why we need this util method

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/util/HiveConfUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    + "<?xml-stylesheet type=\"text/xsl\" href=\"configuration.xsl\"?>\n"
+                    + "<configuration>\n"
+                    + "  <property>\n"
+                    + "    <name>hive.metastore.sasl.enabled</name>\n"
+                    + "    <value>true</value>\n"
+                    + "  </property>\n"
+                    + "</configuration>\n";
+
+    @Test
+    public void test() {
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/util/HiveConfUtilsTest.java:43]: Please give it a more meaningful name

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/util/HiveConfUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        HiveConf hiveConf = createHiveConf();
+        Assert.assertTrue(hiveConf.getBoolVar(HiveConf.ConfVars.METASTORE_USE_THRIFT_SASL));
+
+        // will override configurations from `hiveConf` with hive default values which default value
+        // is null or empty string
+        HiveConf hiveConfNew = new HiveConf(hiveConf, HiveConf.class);
+        Assert.assertFalse(hiveConfNew.getBoolVar(HiveConf.ConfVars.METASTORE_USE_THRIFT_SASL));
+
+        hiveConfNew.addResource(hiveConf);
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/util/HiveConfUtilsTest.java:52]: We should call `HiveConfUtils` to create another new HiveConf and verify the property is not overridden.

**Review by lirui-apache — COMMENTED** [review]
Thanks @dixingxing0 for working on this. I've left some comments.

**Comment by dixingxing0** [comment]
> Thanks @dixingxing0 for working on this. I've left some comments.
Thanks for the comments, I will address them.

**Comment by dixingxing0** [comment]
@lirui-apache I've adressed the comments, please take a look when you have spare time.

**Review by lirui-apache — APPROVED** [review]
Thanks for updating. +1 pending tests.

**Review by wuchong — APPROVED** [review]
LGTM.

**Comment by wuchong** [comment]
@dixingxing0 , could you also create pull requests for release-1.12 and release-1.11?

**Comment by dixingxing0** [comment]
> @dixingxing0 , could you also create pull requests for release-1.12 and release-1.11?

 I would like to do that, i will submit the PR asap.

**Pull request merged**